remote-method/index.js

```diff
@@ -0,0 +1,379 @@
+              value: 'req'
+            },
+            {
+              name: 'store the full response object',
+              value: 'res'
+            },
+            {
+              name: 'store the whole request body',
+              value: 'body'
+            },
+            {
+              name: 'store the whole context object',
+              value: 'context'
+            }
+          ]
+        }
+      ];
+
+      var argName = answers.acceptsArg;
+
+      this.prompt(subprompts, function(answers) {
+        var entry = {
+          arg: argName,
+          type: answers.acceptsType,
+          required: answers.acceptsRequired,
+          description: answers.acceptsDes,
+        };
+        if (answers.httpSource) {
+          entry.http = { source: answers.httpSource };
+        }
+        accepts.push(entry);
+        this.log('\nLet\'s add another accept argument.');
+        this.askForAccepts();
+      }.bind(this));
+    }.bind(this));
+  },
+
+  delimReturns: function() {
+    this.log('Let\'s add some Return Arguments now.\n');
+  },
+
+  askForReturns: function() {
+    var done = this.async();
+    this.log('Enter an empty name when done.');
+
+    var prompts = [
+      {
+        name: 'returnsArg',
+        message: 'Enter name for return argument:',
+        required: true,
+        validate: validateName
+      }
+    ];
+    this.prompt(prompts, function(answers) {
+      if (answers.returnsArg == null || answers.returnsArg === '') {
+        return done();
+      }
+      var subprompts = [
+        {
+          name: 'returnsType',
+          message: 'The type of return argument:',
+          type: 'list',
+          choices: typeChoices
+        },
+        {
+          name: 'returnsRoot',
+          message: 'Is it root',
+          type: 'confirm',
+          default: false
+        },
+        {
+          name: 'returnsDes',
+          message: 'Description of return argument:',
+          required: true
+        }
+      ];
+
+      var argName = answers.returnsArg;
+
+      this.prompt(subprompts, function(answers) {
+        returns.push({
+          arg: argName,
+          type: answers.returnsType,
+          root: answers.returnsRoot,
+          description: answers.returnsDes
+        });
+        this.log('\nLet\'s add another return argument.');
+        this.askForReturns();
+      }.bind(this));
+    }.bind(this));
+  },
+
+  remote: function() {
+    var done = this.async();
+    var def = {
+      name: this.methodName,
+      isStatic: this.isStatic,
+      description: this.description,
+      accepts: accepts,
+      returns: returns,
+      http: https
+    };
+
+    this.modelDefinition.methods.create(def, function(err) {
+      helpers.reportValidationError(err, this.log);
+      return done(err);
+    }.bind(this));
+  },
+
+  buildSampleRemoteMethodSource: function() {
+    var jsFilePath = this.modelDefinition.facetName + '/' +
+    this.modelName + '.js';
```

Comment:
I think it's better to use `path.join`, so that you get `common\my-model.json` on Windows. Thoughts?